sdk/eventhub/event-hubs/samples-dev/sendBufferedEvents.ts

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT Licence.
+
+/**
+ * @summary Demonstrates how to send events to an Event Hub from a buffered producer. This
+ * sample is different from the `sendEvent` one in that it uses the buffered producer
+ * client which buffers events before sending them right away. See the client constructor
+ * options for more information on how to configure the buffering behavior.
+ *
+ * @azsdk-weight 60
+ */
+
+import { AmqpAnnotatedMessage } from "@azure/core-amqp";
+import {
+  EventData,
+  EventHubBufferedProducerClient,
+  OnSendEventsErrorContext,
+} from "@azure/event-hubs";
+
+// Load the .env file if it exists
+import * as dotenv from "dotenv";
+dotenv.config();
+
+const connectionString = process.env["EVENTHUB_CONNECTION_STRING"] || "";
+
+async function handleError(ctx: OnSendEventsErrorContext): Promise<void> {
+  console.log(`The following error occurred:`);
+  console.log(JSON.stringify(ctx.error), undefined, "  ");
+  console.log(
+    `And the following events were not sent as a result to the partition with ID ${ctx.partitionId}:`
+  );
+  console.log(
+    ctx.events.map((event: EventData | AmqpAnnotatedMessage) => JSON.stringify(event)).join("\n\n")
```

Comment:
Just curious: Do we need to type `event` here? Since we are not using any of its properties and are only stringifying it, why does it matter what type it is?

Reply:
Our current configuration of TypeScript disallows untyped parameters. We could type it as `any` but that might not be a good customer experience. We do not access `event` here but the customer might want to and the type annotation could help them to do that right away.